lib/rules/no-invalid-regexp.js

```diff
@@ -154,13 +154,16 @@ module.exports = {
                     return;
                 }
 
-                let flags = getFlags(node);
+                const flags = getFlags(node);
+                let flagsToCheck = flags;
 
-                if (flags && allowedFlags) {
-                    flags = flags.replace(allowedFlags, "");
+                if (flags && allowedFlags && allowedFlags.length > 0) {
+                    allowedFlags.forEach(flag => {
+                        flagsToCheck = flagsToCheck.replace(flag, "");
+                    });
                 }
 
-                let message = validateRegExpFlags(flags);
+                let message = validateRegExpFlags(flagsToCheck);
```

Comment:
Maybe we should pass both the original and the filtered flags to the validation function:

```suggestion
                let message = validateRegExpFlags(flags, flagsToCheck);
```

In `validateRegExpFlags()` we could then validate the `flagsToCheck`, but if the validation fails, we could include all the original flags in the error message.

```js
        function validateRegExpFlags(flags, flagsToCheck) {
            if (!flagsToCheck) {
                return null;
            }
            try {
                validator.validateFlags(flagsToCheck);
            } catch {
                return `Invalid flags supplied to RegExp constructor '${flags}'`;
            }
```

This would retain the current error message in the case of predefined flags, and also remove the inconsistency when the message includes custom flags.

An additional error message for duplicated flags would be nice, but that could be added as a feature in another PR.

Reply:
sounds good, but how about if we only show those flags in error message that are invalid? i think that can be helpful to find which flags we need to remove.

Reply:
> sounds good, but how about if we only show those flags in error message that are invalid? i think that can be helpful to find which flags we need to remove.

Makes sense, as long as we also show duplicated flags.